app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt

```
@@ -122,8 +135,48 @@ class SignUpViewModel(private val companyRepository: CompanyRepository) : ViewMo
+                    } else {
+                        val errorBody = response.errorBody()!!.string()
+                        println("errorBody $errorBody")
+                        val jsonObject = JSONObject(errorBody)
+                        println("jsonObject $jsonObject")
+                        val message = jsonObject.getString("message")
+                        println("message $message")
+
+                        ErrorUIState.Error(message = message)
+                    }
+                }
+            } catch (e: IOException) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)